tests/Run-FunctionalTests.ps1

```diff
@@ -176,7 +182,31 @@ function ExecTest($argList)
     else
     {
         $arguments = "$testPath" + "$argList"
-        Invoke-Expression "$taefPath $arguments"
+        if($RedirectTAEFErrors) {
+            $output = Invoke-Expression "$taefPath $arguments"
+            $script:exitCode = $LASTEXITCODE
+            foreach($o in $output) {
+                            
+                if($o.StartsWith("Error:") -or $o.Contains("[Failed]")) {
+
+                    #TAEF does not report exceptions/crashes/failures during test cleanup as test failures.
+                    #However, it will log a message like: "Error: TAEF: [HRESULT 0x8000FFFF] ...."
+                    #We want to detect and mark these as test failures.
+                    $script:exitCode = 1
+
+                    #Note: Adding ##[error]: is just to make the errors show up as red in VSTS output
+                    #Unfortunately, writing the output to stderr (write-error) creates a lot of noise in powershell.
+                    #And attempting to use Console.WriteError or $host.ui.WriteErrorLine results in lines be displayed out of order.
+                    write-host "##[error]: $o" -foregroundcolor red
+
+                } else {
+                    write-host $o
+                }
+            }
+        } else {
+            Invoke-Expression "$taefPath $arguments"
+            $script:exitCode = $LASTEXITCODE
+        }
```

Comment:
Brian should just be able to take this and add it to his pull request so we can have the UT's running TAEF as well.